src/Features/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -27,5 +32,23 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
                 return nonMutatingRequestTask.ReportNonFatalErrorAsync();
             }
         }
+
+        private void TrySetLocale()
+        {
+            var locale = _initializeManager.TryGetInitializeParams()?.Locale;
+            // The client may not have given us a UI or this is the initialize request and we haven't saved it yet.
+            if (!string.IsNullOrWhiteSpace(locale))
+            {
+                try
+                {
+                    var desiredUICulture = CultureInfo.CreateSpecificCulture(locale);
+                    CultureInfo.CurrentUICulture = desiredUICulture;
+                }
+                catch (CultureNotFoundException)
+                {
+                    _logger.LogWarning($"Culture {locale} was not found, falling back to OS culture");
```

Comment:
If the client supports more languages than the server, how will `CultureInfo.CreateSpecificCulture(locale);` behaved? Will it fall back the OS default language?

Reply:
Maybe add a unit test to this scenario?

Reply:
If it doesn't exist in `CultureInfo.CreateSpecificCulture(locale)`, it will throw.  If it exists but we don't have a translation, it will fall back

Reply:
Added test for when the locale is invalid